docs/elasticsearch/index-data.md

```diff
@@ -78,3 +77,146 @@ Elasticsearch indices have the following naming restrictions:
 - Index names can't contain spaces, commas, or the following characters:
 
   `:`, `"`, `*`, `+`, `/`, `\`, `|`, `?`, `#`, `>`, or `<`
+
+## Read data
+
+After you index a document, you can retrieve it by sending a GET request to the same endpoint that you used for indexing:
+
+```json
+GET movies/_doc/1
+
+{
+  "_index": "movies",
+  "_type": "_doc",
+  "_id": "1",
+  "_version": 1,
+  "found": true,
+  "_source": {
+    "title": "Spirited Away"
+  }
+}
+```
+
+You can see the document in the `_source` object. If the document is not found, the `found` key will be set to `false` and the `_source` object will not be part of the result.
+
+To retrieve multiple documents with a single command, use the `_mget` operation.
+The format for retrieving multiple documents is similar to the `_bulk` operation, where you must specify the index and ID in the request body:
+
+```json
+GET _mget
+{
+  "docs": [
+    {
+      "_index": "<index>",
+      "_id": "<id>"
+    },
+    {
+      "_index": "<index>",
+      "_id": "<id>"
+    }
+  ]
+}
+```
+
+To only return specific fields in a document:
+
+```json
+GET _mget
+{
+  "docs": [
+    {
+      "_index": "<index>",
+      "_id": "<id>",
+      "_source": "field1"
+    },
+    {
+      "_index": "<index>",
+      "_id": "<id>",
+      "_source": "field2"
+    }
+  ]
+}
+```
+
+To check if a document exists:
+
+```json
+HEAD movies/_doc/<doc-id>
+```
+
+If the document exists, you get back a `200 OK` response, and if it doesn't, you get back a `404 - Not Found` error.
+
+## Update data
+
+To update existing fields or to add new fields, send a POST request to the `_update` operation with your changes in a `doc` object:
+
+```json
+POST movies/_doc/1/_update
+{
+  "doc": {
+    "title": "Castle in the Sky",
+    "genre": ["Animation", "Fantasy"]
+  }
+}
+```
+
+The `title` field is updated and the `genre` field is added:
+
+```json
+GET movies/_doc/1
+
+{
+  "_index": "movies",
+  "_type": "_doc",
+  "_id": "1",
+  "_version": 2,
+  "found": true,
+  "_source": {
+    "title": "Castle in the Sky",
+    "genre": [
+      "Animation",
+      "Fantasy"
+    ]
+  }
+}
+```
+
+Note that the `_version` field of the document is incremented to 2. Use the `_version` field to keep track of how many times a document is updated.
+
+POST requests are made for partial updates to a document. To altogether replace a document, use a PUT request:
+
+```json
+PUT movies/_doc/1
+{
+  "title": "Spirited Away"
+}
+```
+
+The document with ID of 1 will contain only the `title` field, because the entire document will be replaced with the document indexed in this PUT request.
+
+You can conditionally update a document based on whether or not it already exists using the `upsert` object.
+Here, if the document exists, it's `title` field is changed to `Castle in the Sky`; if it doesn't already exist, the document within the `upsert` object is indexed:
```

Comment:
Use the `upsert` object to conditionally update documents based on whether they already exist. Here, if the document exists, its title field changes to `Castle in the Sky`. If it doesn't, Elasticsearch indexes the document in the `upsert` object.